packages/smart-vaults/src/signers/Signer.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.23;
+
+/**
+ * @notice An EOA or Passkey signer.
+ *
+ * @dev For a Signer to be valid it has to be either an EOA or a Passkey signer.
+ *      - EOA -> y has to be empty and x has to be a valid address.
+ *      - Passkey -> both x and y have to be non empty.
+ */
+struct Signer {
```

Comment:
using x and y is a little confusing bc of the overlap w passkey pub keys (vs something more generic like slot0 and slot1) but i think it's probably fine

Reply:
good shout